code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -0,0 +1,310 @@
+	..()
+	processing_objects.Add(src)
+
+/obj/item/weapon/gun/energy/awaymission_aeg/Destroy()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/item/weapon/gun/energy/awaymission_aeg/process()
+	var/turf/my_loc = get_turf(src)
+	if(my_loc.z == (MAX_Z + 1))
+		if (inawaymission)
+			charge_tick++
+			if(charge_tick < 4) return 0
+			charge_tick = 0
+			if(!power_supply) return 0
+			if((power_supply.charge / power_supply.maxcharge) != 1)
+				power_supply.give(1000)
+				update_icon()
+		else
+			to_chat(get(src, /mob), "<span class='notice'>Your [src] activates, starting to draw power from a nearby wireless power source.</span>")
+			inawaymission = 1
+	else
+		if (inawaymission)
+			to_chat(get(src, /mob), "<span class='danger'>Your [src] deactivates, as it is out of range from its power source.</span>")
+			power_supply.charge = 0
+			inawaymission = 0
+			update_icon()
+
+/obj/item/weapon/gun/energy/awaymission_aeg/proc/update_charge()
+	var/ratio = power_supply.charge / power_supply.maxcharge
+	ratio = round(ratio, 0.25) * 100
+	overlays += "nucgun-[ratio]"
+
+/obj/item/weapon/gun/energy/awaymission_aeg/proc/update_reactor()
+	if ((power_supply.charge/power_supply.maxcharge) <= 0.5)
+		overlays += "nucgun-light"
+	else
+		overlays += "nucgun-clean"
+
+/obj/item/weapon/gun/energy/awaymission_aeg/proc/update_mode()
+	overlays += "nucgun-kill"
+
+/obj/item/weapon/gun/energy/awaymission_aeg/emp_act(severity)
+	..()
+	reliability = max(reliability - round(15/severity), 0) //Do not allow it to go negative!
+
+/obj/item/weapon/gun/energy/awaymission_aeg/update_icon()
+	overlays.Cut()
+	update_charge()
+	update_reactor()
+	update_mode()
+
+/obj/item/weapon/gun/energy/awaymission_aeg/attack_self(mob/living/user as mob)
+	to_chat(user, "<span class='danger'>[name] appears to only have one setting, scrawled hastily on it in pen: 'SPIDERS!!!'. This is probably a bad sign.</span>")
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin
+	name = "beaker 'Black Terror Venom'"
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin/New()
+	..()
+	reagents.add_reagent("terror_black_toxin", 50)
+	update_icon()
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_green_toxin
+	name = "beaker 'Green Terror Pheromone'"
+
+/obj/item/weapon/reagent_containers/glass/beaker/terror_black_toxin/New()
+	..()
+	reagents.add_reagent("terror_green_toxin", 50)
+	update_icon()
+
+/obj/item/weapon/gun/energy/awaymission_spidergun
+	name = "Modified Decloner"
+	desc = "A modified biological decloner, tweaked to destroy terror spiders more effectively."
+	icon_state = "decloner"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+	origin_tech = "combat=5;materials=4;powerstorage=3"
+	projectile_type = "/obj/item/projectile/energy/declone_spider"
+
+/obj/item/projectile/energy/declone_spider
+	name = "declone"
+	icon_state = "declone"
+	damage = 20
+	damage_type = CLONE
+
+/obj/machinery/computer/id_upgrader
```

Comment:
If the first door is hacked, isn't that enough to prevent them from accessing the rest of them, unprepared? Likewise, if it's just left open, those who aren't prepared, but want to still look around are just going to shove their ID in this machine anyway. There's nothing inherently wrong with this, it's just--having tons of unique machinery on an away mission for something minor like this seems...well, redundant.